File: WordApp/ClientApp/src/app/tasks/task.model.ts
export class TaskModel {
  name: string;
  taskType: string;
  id: string;
  rowVersion: string;

  constructor(name: string, type: string, id: string, rowVersion: string) {
    this.name = name;
    this.taskType = type;
    this.id = id;
    this.rowVersion = rowVersion;
  }
}

File: WordApp/Models/TaskModel.cs
using Entities.Enums;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace WordApp.Models
{
    public class TaskModel : BaseModel
    {
        public string Name { get; set; }
        [JsonConverter(typeof(StringEnumConverter))]
        public TaskType TaskType { get; set; }
    }
}
